.github/workflows/check-and-gen.yml

```diff
@@ -70,3 +68,38 @@ jobs:
       - uses: ./.github/actions/install-go
       - run: go mod tidy -v
       - run: git diff --exit-code
+  generate-docs:
+    name: Generate Documentation
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/heads/release/'))
```

Comment:
I'm unsure about this and whether a merge into one of these branches would trigger the job as well, we only want it to run on the PR branch, strictly. So this line worries me a little and it needs verification, somehow, or we need a subject-matter expert to help.

Reply:
@BigLep maybe something to pull IPDX into to help review.

Reply:
@galargh : do you know offhand whether this will only run on pushes to PR branches and not on master or release branches?

Reply:
That's the opposite of what this conditional says.

The conditional is: accept a workflow run which was triggered by a push event to either `master` or `release/*` branch.